content/post/next_compiler.md

```diff
@@ -0,0 +1,207 @@
+happened many times before in the history.
+
+## Feedback
+
+Programming languages evolve, as the natural ones do. The rules of a natural language reflect
+the norms which are adopted by the majority of the speakers. There is evidence that the complexity
+of a natural language decreases over time. If you compare Russian language from 19th century
+to the one in 21st, you will surprise how ridiculously complex is the former compared to the
+latter: more letters, more rules, more exceptions. Funnily enough, we do not know any programming
+language which does becomes simpler or easier to learn over time.
+
+There are several reasons we can imagine for that. While in a natural language words are either
+dropped or replaces, programming languages keep gaining new functions yet at the same time do
+not remove old ones and maintain backwards compatibility. The sad example of Python 3 clearly
+shows that revolutionary changes harm so much that they can split the community.
+
+Francesc, the hed of our Developer Relations, presented a curious example of Go language evolution
+in his [post on Medium](https://medium.com/@francesc/source-d-why-i-left-google-71240c0f0204).
+The community ran the analysis over the existing open source projects and
+[proved](https://github.com/golang/go/issues/14595) the usefulness of a new function in the
+standard library. It demonstrates how the programming languages develop today: people give
+explicit feedback (RFCs, JSRs, PEPs, GitHub issues, etc.) and the owners decide on them and
+sometimes fulfill the requests. The time to live is months, if not years. Let's consider how
+we are able change it by applying ML on "big code".
+
+We are not being extreme at first. We can run [pattern mining](https://arxiv.org/pdf/1602.05012.pdf)
+to further automate the proposals process. There are multiple ways to mine, will it be a
+recurrent neural network with attention and pattern memory or anything else. It also depends on
+the target pattern shape: fixed and rigid or flexible and template-like small or big.
+We bet that the following code will be #1 top recurring snippet in Go:
+
+```go
+..., err := /* whatever func() */
+if err != nil {
+  panic(err)  // or [return] whatever func(err)
+}
+```
+
+Examples: [here](https://gobyexample.com/reading-files)
+or [here](https://github.com/divan/gotrace/blob/master/main.go#L38)
+or [here](https://github.com/moby/moby/blob/master/client/client.go#L141)
+or [here](https://github.com/src-d/go-git/blob/master/_examples/common.go#L19)
+(We can easily continue this for a long time). Yet this is a feature of the initial language design
+and no way its owners are doing anything with it. This is an important observation: even if the
+pattern is statistically obvious, it may be rejected by a tiny group of people who invented
+the language and are maintaining it. By no means we blame anyone; this is their right.
```

Comment:
Please, do not write this. Attacking the maintainers of an open source project should not be our style.